src/figuresClasses.ts

```diff
@@ -1,11 +1,70 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: string = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('any length <= 0');
+    }
+
+    if (a >= b + c || b >= a + c || c >= b + a) {
+      throw new Error('longest side of a triangle >= sum of other two sides');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  public shape: string = 'circle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('any length <= 0');
+    }
+  }
 
-export class Triangle implements Figure {}
+  getArea(): number {
+    return Math.floor(Math.PI * Math.pow(this.radius, 2) * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  public shape: string = 'rectangle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('any length <= 0');
+    }
+  }
 
-export class Rectangle implements Figure {}
+  getArea(): number {
+    return Math.floor((this.width * this.height * 100) / 100);
```

Comment:
The `getArea` method in the `Rectangle` class should correctly apply `Math.floor` to the entire area calculation. The current implementation mistakenly divides by 100 before applying `Math.floor`, which may lead to incorrect results. Consider enclosing the multiplication in parentheses: `Math.floor(this.width * this.height * 100) / 100`.